cli/cliui/table.go

```diff
@@ -41,3 +44,174 @@ func FilterTableColumns(header table.Row, columns []string) []table.ColumnConfig
 	}
 	return columnConfigs
 }
+
+// DisplayTable renders a table as a string. The input argument must be a slice
+// of structs (panics otherwise, even for empty slices). At least one field in
+// the struct must have a `table:""` tag containing the name of the column in
+// the outputted table.
+//
+// Nested structs are processed if the field has the `table:"$NAME,recursive"`
+// tag and their fields will be named as `$PARENT_NAME $NAME`. If the tag is
+// malformed or a field is marked as recursive but does not contain a struct or
+// a pointer to a struct, this function will panic (even with an empty input
+// slice).
+//
+// If sort is empty, the input order will be used. If filterColumns is empty or
+// nil, all available columns are included.
+func DisplayTable(out interface{}, sort string, filterColumns []string) (string, error) {
+	v := reflect.Indirect(reflect.ValueOf(out))
+
+	if v.Kind() != reflect.Slice {
+		panic("DisplayTable called with a non-slice type")
+	}
+
+	headersRaw := typeToTableHeaders(v.Type().Elem())
+	if len(headersRaw) == 0 {
+		panic("no table tags found on the input type")
+	}
+	headers := make(table.Row, len(headersRaw))
+	for i, header := range headersRaw {
+		headers[i] = header
+	}
+
+	tw := Table()
+	tw.AppendHeader(headers)
+	tw.SetColumnConfigs(FilterTableColumns(headers, filterColumns))
+	tw.SortBy([]table.SortBy{{
+		Name: sort,
+	}})
+
+	// Write each struct to the table.
+	for i := 0; i < v.Len(); i++ {
+		// Format the row as a slice.
+		rowMap := valueToTableMap(v.Index(i))
+		rowSlice := make([]interface{}, len(headers))
+		for i, h := range headersRaw {
+			v, ok := rowMap[h]
+			if !ok {
+				v = nil
+			}
+
+			// Special type formatting.
+			switch val := v.(type) {
+			case time.Time:
+				v = val.Format(time.Stamp)
+			case *time.Time:
+				if val != nil {
+					v = val.Format(time.Stamp)
+				}
+			}
+
+			rowSlice[i] = v
+		}
+
+		tw.AppendRow(table.Row(rowSlice))
+	}
+
+	return tw.Render(), nil
+}
+
+func parseTableStructTag(field reflect.StructField) (name string, recurse bool, ok bool) {
+	tag, ok := field.Tag.Lookup("table")
+	if !ok {
+		return "", false, false
+	}
+
+	tagSplit := strings.Split(tag, ",")
```

Comment:
Suggestion (non-blocking): If you want to get rid of the manual tag parsing, [fatih/structtag](https://github.com/fatih/structtag) is a pretty awesome lib.